internal/cmd/run.go

```diff
@@ -89,16 +88,26 @@ func runCmd() *cobra.Command {
 					for _, task := range tasks {
 						block := task.CodeBlock
 
-						if runAll && block.ExcludeFromRunAll() {
+						if runAll && len(categories) == 0 && block.ExcludeFromRunAll() {
 							continue
 						}
 
-						if category != "" {
+						if len(categories) > 0 {
 							if block.ExcludeFromRunAll() {
 								continue
 							}
 
-							if !slices.Contains(strings.Split(block.Category(), ","), category) {
+							bcats := strings.Split(block.Category(), ",")
```

Comment:
Suggestion: Add `block.Categories()` and mark `block.Category()` as deprecated using `// Deprecated: Use Categories() instead`.